### PR TITLE
Reactivating network mtclient (Parallax Online / PXO)

### PR DESCRIPTION
**Pull request by winterheart**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [x] Network changes
  - [ ] Other changes

### Description

Reactivating from legacy mtclient in order to enable PXO online service. Currently it's able to login tracker server with valid credentials from https://pxo.nottheeye.com/ (but unable to initialize chat server connection, need additional connection settings - server and port).

### Related Issues

This PR is part of #216.

### Screenshots (if applicable)

Screenshot_20240603_164749
Screenshot_20240603_164803
Screenshot_20240603_165023

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

File: netcon/mtclient/chat_api.cpp [hunk 1]
```diff
@@ -141,16 +140,10 @@ int ConnectToChatServer(char *serveraddr, char *nickname, char *trackerid) {
     Chat_server_connected = 0;
     FlushChatCommandQueue();
 
-    p = strchr(serveraddr, ':');
+    strncpy(chat_server, serveraddr, strlen(serveraddr));
+    chat_server[strlen(serveraddr)] = '\0';
```
Comment (Lgt2x) [inline netcon/mtclient/chat_api.cpp:144]: can be a problem if `serveraddr` is longer than `chat_server`

File: netcon/mtclient/mtclient.h [hunk 2]
```diff
@@ -80,6 +80,9 @@
 
 #include "ui.h"
 
+// Chat tracker
+#define CHATTRACKERNAME "tracker.pxo.nottheeye.com"
+#define CHATPORT 7170
```
Comment (Lgt2x) [inline netcon/mtclient/mtclient.h:85]: should we hardcode these values?

File: netcon/mtclient/mtgametrack.h [hunk 3]
```diff
@@ -96,8 +96,7 @@
 #pragma pack(1)
 
 // Definitions
-#define GAMETRACKER "209.223.64.251" // The scanner machine at Outrage
-#define GAMETRACKERNAME "gt.parallaxonline.com"
+#define GAMETRACKERNAME "tracker.pxo.nottheeye.com"
```
Comment (Lgt2x) [inline netcon/mtclient/mtgametrack.h:99]: `#define GAMETRACKERNAME CHATTRACKERNAME` if it is usually the same

File: netcon/inetfile/Chttpget.cpp [hunk 4]
```diff
@@ -211,7 +211,12 @@ void ChttpGet::PrepSocket(char *URL) {
     return;
   }
 
-  make_nonblocking(m_DataSock);
```
Comment (Lgt2x) [inline netcon/inetfile/Chttpget.cpp:214]: make_nonblocking fails here? I thought we fixed it

File: netcon/mtclient/mtclient.h (hunk 2 above)
Comment (winterheart) [reply netcon/mtclient/mtclient.h:85]: This is first iteration, in future we'll extend these strings to configuration.

File: netcon/mtclient/mtgametrack.h (hunk 3 above)
Comment (winterheart) [reply netcon/mtclient/mtgametrack.h:99]: There three defines - GAMETRACKER, CHATTRACKER and PLAYERTRACKER. In this case they are identical, but can be different. I'd prefer to keep them separated.

File: netcon/inetfile/Chttpget.cpp (hunk 4 above)
Comment (winterheart) [reply netcon/inetfile/Chttpget.cpp:214]: `make_nonblocking()` pulls into DLL unnecessary dependencies - networking, ddio and friend of these.

File: netcon/mtclient/mtclient.h (hunk 2 above)
Comment (Lgt2x) [reply netcon/mtclient/mtclient.h:85]: yeah, registry/config file seems like a good fit for them

File: netcon/inetfile/Chttpget.cpp (hunk 4 above)
Comment (Lgt2x) [reply netcon/inetfile/Chttpget.cpp:214]: got it, maybe make_nonblocking should go somewhere more accessible

**Review by Lgt2x — APPROVED** [review]
Looking good!

**Pull request merged**